flexmeasures/api/v3_0/sensors.py

```diff
@@ -106,8 +127,75 @@ def index(self, account: Account):
         :status 403: INVALID_SENDER
         :status 422: UNPROCESSABLE_ENTITY
         """
-        sensors = get_sensors(account=account)
-        return sensors_schema.dump(sensors), 200
+        accounts: list = [account] if account else []
+        account_ids: list = [acc.id for acc in accounts]
+
+        if all_accessible is not None:
+            consultancy_account_ids: list = [
+                acc.consultancy_account_id for acc in accounts
+            ]
+            account_ids.extend(consultancy_account_ids)
+
+        filter_statement = GenericAsset.account_id.in_(account_ids)
+
+        sensor_query = (
+            select(Sensor)
+            .join(GenericAsset, Sensor.generic_asset_id == GenericAsset.id)
+            .join(Account, GenericAsset.owner)
+            .filter(
+                or_(
+                    filter_statement,
+                    GenericAsset.account_id.is_(None),
```

Comment:
We should only add public assets when not just one specific account was requested.

Reply:
I’m not quite sure I understand what you mean here. Are you saying that public assets should only be included if we’re looking up more than one account?

Reply:
Hm, better to say we get public assets when all_accessible is True